cosmos.go

```diff
@@ -295,40 +303,49 @@ func (c *cosmosImpl) retryLoop(executeRequest retryFun) (responses []interfaces.
 		}
 
 		if retryInformation.retryAfter > 0 {
-			c.logger.Info().Msgf("retry %d of query after %v because of header status code %d", tryCount+1, retryInformation.retryAfter, retryInformation.responseStatusCode)
+			logger.Info().Msgf("retry %d of query after %v because of header status code %d", tryCount+1, retryInformation.retryAfter, retryInformation.responseStatusCode)
 
 			if waitDone := waitForRetry(retryInformation.retryAfter, timeoutReachedChan); !waitDone {
-				break
+				// timeout occurred
+				metrics.requestRetryTimeoutsTotal.Inc()
+				return nil, fmt.Errorf("timed out while waiting to do a retry after %s (timeout=%s)", retryInformation.retryAfter, retryTimeout)
 			}
 		}
 
 		// Timeout check in case no waiting is required
 		select {
 		case <-timeoutReachedChan:
 			// we stop here and return what we got so far
-			return responses, err
+			metrics.requestRetryTimeoutsTotal.Inc()
+			return nil, fmt.Errorf("timed out while doing a retry (timeout=%s)", retryTimeout)
```

Comment:
not sure if this is the better solution, the first answer received would still be more valuable, even if it failed due to throttling or whatever

Reply:
- will move the logmessage 
- won't return an error 
- will return the responses